### PR TITLE
Added peds to shops

### PR DESCRIPTION
**Pull request by TonybynMp4**
It's probably really badly implemented so don't hesitate to review with changes (like please do it)!
### know bug/problem:
You can't have targets AND peds for the same 'type' of shop
it won't spawn the peds because of the elseif (that's an easy fix buuuut....) 
if they did spawn (with the targets) then you can't open the ped's shops (i think the shop doesn't get added to the Shops table or something ? )
I tried changing the id's to i +100 when it's peds and it just never appears in the Shops table 🤷🏻

### CODE REVIEW

File: modules/shops/client.lua [hunk 1]
```diff
@@ -106,6 +106,46 @@ client.shops = setmetatable(data('shops'), {
...
+
+							shops[id] = {
+								zoneId = shopid,
+								blip = blip and createBlip(blip, ped.loc)
+							}
+							local pedmodel = joaat(ped.ped)
+							RequestModel(pedmodel)
+							while not HasModelLoaded(pedmodel) do
+								Wait(0)
+							end
```
Comment (FjamZoo) [inline modules/shops/client.lua:123]: There's a ox_lib function to do these makes it look cleaner

File: modules/shops/client.lua [hunk 2]
```diff
@@ -106,6 +106,46 @@ client.shops = setmetatable(data('shops'), {
...
+							local PedId = CreatePed(0, ped.ped, ped.loc.x, ped.loc.y, ped.loc.z-1, ped.loc.w, true, true)
```
Comment (FjamZoo) [inline modules/shops/client.lua:124]: Why is the peds here Networked? that means if multiple people join shops will be messed up

Comment (TonybynMp4) [reply modules/shops/client.lua]: cause i'm dumb sorry x)

**Comment by thelindat** [comment]
iirc, shops will be recreated when your "groups" change (jobs, gangs, whatever) which will lead to extra peds getting spawned in; also no handling for resource restart?

Really the handling for points and zones needs to be redone entirely, then peds added in; also requiring support for older format so people don't need to redo everything when updating.

**Comment by TonybynMp4** [comment]
> iirc, shops will be recreated when your "groups" change (jobs, gangs, whatever) which will lead to extra peds getting spawned in;

indeed that definitely needs to be fixed somehow

> also no handling for resource restart?

Is it really necessary? the peds automatically go away (/ disappears it's kinda bugged i guess) and aren't targetable anymore, i don't think it really needs a restart function 🤔 

> Really the handling for points and zones needs to be redone entirely, then peds added in; also requiring support for older format so people don't need to redo everything when updating. This would ideally also drop support for qtarget as previously planned.

That sounds like a painful thing to do, i could probably replace the qtarget exports for ox_target to start if you want.

**Comment by thelindat** [comment]
> > iirc, shops will be recreated when your "groups" change (jobs, gangs, whatever) which will lead to extra peds getting spawned in;
> 
> indeed that definitely needs to be fixed somehow

Rather than fixing this particular problem, rewriting everything would solve it.

> > also no handling for resource restart?
> 
> Is it really necessary? the peds automatically go away (/ disappears it's kinda bugged i guess) and aren't targetable anymore, i don't think it really needs a restart function 🤔

They'll stop being handled by the script and go do regular npc stuff; still weird and annoying.

> > Really the handling for points and zones needs to be redone entirely, then peds added in; also requiring support for older format so people don't need to redo everything when updating. This would ideally also drop support for qtarget as previously planned.
> 
> That sounds like a painful thing to do, i could probably replace the qtarget exports for ox_target to start if you want.

Yes, trying to upgrade things is a nuisance; more-so when you also need to keep compatibility.

**Comment by TonybynMp4** [comment]
> Rather than fixing this particular problem, rewriting everything would solve it.

> They'll stop being handled by the script and go do regular npc stuff; still weird and annoying.

i think i figured out how to fix it temporarily, peds despawn at resource stop and when re-creating shops.
I wish i could help you rewrite the entire shop system but that's definitely outside of my capacities sadly :')

I also converted a few qtarget exports to ox_target.
I noticed the evidence and stashes would be annoying to convert so i'll try later today

**Comment by thelindat** [comment]
I recommend waiting for the time being, as I said before
> the handling for points and zones needs to be redone entirely

This'll be a priority for the next release in order to resolve #422, #960, #1034

**Comment by thelindat** [comment]
I've rewritten this crap nearly a dozen times now (wish I was exaggerating). I'm so unhappy with the overall structure for shops but it's not something I can just change because of backwards compatibility.

Probably going to have to be something to come with inventory v3, but my motivation to work on this stuff has been massively lacking for the past ~3 months.

**Comment by TonybynMp4** [comment]
i think i fixed the issues there 🤔 

> This'll be a priority for the next release in order to resolve #422, #960, #1034

**Comment by thelindat** [comment]
I’ve added ped spawning in without any of the other much-needed changes that shops need; mostly because backwards compatibility is too much of a hassle and making some temporary tweaks just to eventually overhaul things in a major update leaves a sour taste.

Peds are loaded alongside zones and use the same table/data, with the addition of adding ped and scenario fields.

Making zones work with markers is an issue because they conflict with the current setup, and people updating will suddenly find themselves loading a set of shops they don’t want.

**Pull request closed without merging**